src/iso_erlang.erl

```diff
@@ -199,12 +219,26 @@ language_alpha_2_to_name(Language) when is_list(Language) orelse is_bitstring(La
 -spec language_alpha_3_to_name(bitstring() | string()) -> bitstring().
 language_alpha_3_to_name(Language) when is_list(Language) orelse is_bitstring(Language) -> iso_erlang_language_converter:alpha_3_to_name_lower(to_lower(Language)).
 
+-spec to_upper(bitstring() | string()) -> bitstring().
 to_upper(String) when is_list(String) ->
     list_to_bitstring(string:to_upper(String));
 to_upper(Bitstring) when is_bitstring(Bitstring) ->
     <<<<(string:to_upper(X))>> || <<X>> <= Bitstring>>.
 
+-spec to_lower(bitstring() | string()) -> bitstring().
 to_lower(String) when is_list(String) ->
     list_to_bitstring(string:to_lower(String));
 to_lower(Bitstring) when is_bitstring(Bitstring) ->
-    <<<<(string:to_lower(X))>> || <<X>> <= Bitstring>>.
\ No newline at end of file
+    <<<<(string:to_lower(X))>> || <<X>> <= Bitstring>>.
+
+-spec normalize_language(bitstring() | string()) -> bitstring() | error.
+normalize_language(Language) when is_bitstring(Language) orelse is_list(Language) ->
+    remove_dialect(to_lower(Language)).
+
+-spec remove_dialect(bitstring()) -> bitstring() | error.
+remove_dialect(<<Alpha2:16/bitstring>>) -> Alpha2;
+remove_dialect(<<Alpha3:24/bitstring>>) -> Alpha3;
+remove_dialect(<<Alpha2:16/bitstring, $-, _Rest/bitstring>>) -> Alpha2;
```

Comment:
I could get behind that. Just wanted to make sure that I understood it the same way you intended it.